spec/integration/mixed_collection_spec.rb

```diff
@@ -0,0 +1,59 @@
+require 'spec_helper'
+
+RSpec.describe JSONAPI::Serializer do
+  let(:car) { Car.fake }
+  let(:bus) { Bus.fake }
+  let(:truck) { Truck.fake }
+
+  context 'when serializing a mixed collection' do
+    it 'uses the correct serializer for each object' do
+      vehicles = VehicleSerializer.new([car, bus], serializers: { Car: CarSerializer, Bus: BusSerializer }).to_hash
```

Comment:
To be honest, I'm not sure how OK is to always have to pass the serializers inline. I'd rather memoize the `serializer_for` and cache calls to it...

Reply:
What do you mean here?

When I first wrote this I went back and forth about how to define what serializers to use where but couldn't come up with a better way to do this within the existing class structure. If you have a better idea, by all means!